nexus_constructor/nexus/nexus_wrapper.py

```diff
@@ -187,6 +187,24 @@ def get_field_value(group: h5py.Group, name: str) -> Optional[Any]:
             value = str(value, "utf8")
         return value
 
+    def get_fields(self, group: h5py.Group) -> List[Union[h5py.Dataset, h5py.Group]]:
+        fields = []
+        for item in group.values():
+            if (
+                isinstance(item, h5py.Dataset)
+                and item.name.split("/")[-1] != "description"
+            ):
+                fields.append(item)
+            if isinstance(item, h5py.Group):
+                if isinstance(item.parent.get(item.name, getlink=True), h5py.SoftLink):
+                    fields.append(item)
+                if (
+                    "NX_class" in item.attrs.keys()
+                    and item.attrs["NX_class"] == "NCstream"
+                ):
+                    fields.append(item)
+        return fields
```

Comment:
The filtering this function does is basically the same as the filtering done in `add_component_window` can we avoid doing it twice?